File: server/model/user_model.js
const mongoose = require('mongoose')

const { Schema } = mongoose

const UserSchema = new Schema({
    fullName: { type: String, required: true },
    phone: { type: String, required: true },
    password: {type: String,required: true},
    email: { type: String, required: true, unique: true },
    city: { type: String },
    fcmToken: {type: String,required: true},
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
  });

  const UserModel = mongoose.model('User',UserSchema)
  module.exports=UserModel

File: mobile/lib/model/user_model.dart
import 'dart:convert';

class UserModel {
  UserModel({
    required this.id,
    required this.fullName,
    required this.email,
    required this.phone,
    required this.city,
  });

  String id;
  String fullName;
  String email;
  String phone;
  String city;

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      '_id': id,
      'fullName': fullName,
      'email': email,
      'phone': phone,
      'city': city,
    };
  }

  factory UserModel.fromMap(Map<String, dynamic> map) {
    return UserModel(
      id: map['_id'] as String,
      fullName: map['fullName'] as String,
      email: map['email'] as String,
      phone: map['phone'] as String,
      city: map['city'] as String,
    );
  }
}
